### PR TITLE
Improve GA Workflow

### LINKED ISSUE #1972: Scheduled workflow failed
Oh no, something went wrong in the scheduled workflow **PyTorch version tests with commit edc418353e11913aaeee0481654fc25c8f6cd1b2**.
Please look into it:

https://github.com/pytorch/ignite/actions/runs/787371309

Feel free to close this if this was just a one-off error.

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #1972 

Added downloading MNIST to `pytorch-versions-tests.yml` and moved downloading MNIST before running the tests in `unit-tests.yml` and copied it.

Thanks to @trsvchn for helping me with each step in this PR!

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Review by trsvchn — APPROVED** [review]
Thanks @KickItLikeShika ! LGTM!

**Comment by KickItLikeShika** [comment]
Are these failures related to out updates?

**Comment by trsvchn** [comment]
> Are these failures related to out updates? everything went fine on my fork https://github.com/KickItLikeShika/ignite/actions/runs/797863425 ...
> 

No, I don't think so, smth wrong with cache action
```
Warning: Cache service responded with 429 during upload chunk.
D:\a\_actions\actions\cache\v2\dist\save\index.js:4043
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`)
```

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -91,6 +91,17 @@ jobs:
...
+      # Download MNIST: https://github.com/pytorch/ignite/issues/1737
+      - name: Download MNIST
+        uses: pytorch-ignite/download-mnist-github-action@master
+        with:
+          target_dir: /tmp
+
+      - name: Copy MNIST
+        shell: bash -l {0}
+        run: |
+          cp -R /tmp/MNIST .
```
Comment (vfdev-5) [inline .github/workflows/unit-tests.yml:103]: why do we need to copy it ?

Comment (trsvchn) [reply .github/workflows/unit-tests.yml:103]: We copy it for examples, do you think refactor examples would be better?

Comment (vfdev-5) [reply .github/workflows/unit-tests.yml:103]: Please, put a comment explaining that

**Pull request merged**